src/main/java/core/basesyntax/strategy/StorageReportGenerateImpl.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import java.util.Map;
+
+public class StorageReportGenerateImpl implements StorageReportGenerate {
+    public String getReport() {
+        StringBuilder report = new StringBuilder();
+        report.append("fruit,quantity").append(System.lineSeparator());
+        for (Map.Entry<String, Integer> entry : Storage.getAllFruits().entrySet()) {
```

Comment:
Direct access to the `Storage` class's static method `getAllFruits()` from within `StorageReportGenerateImpl` violates the encapsulation principle. Instead, consider passing the storage data as a parameter to the `getReport` method or use a service that provides the necessary data.